src/Aspire.Hosting/api/Aspire.Hosting.cs

```diff
@@ -2603,40 +3342,42 @@ public enum ContainerImageFormat
     }
 
     [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPIPELINES003", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    [System.Flags]
     public enum ContainerTargetPlatform
     {
-        LinuxAmd64 = 0,
-        LinuxArm64 = 1,
-        LinuxArm = 2,
-        Linux386 = 3,
-        WindowsAmd64 = 4,
-        WindowsArm64 = 5
+        LinuxAmd64 = 1,
+        LinuxArm64 = 2,
+        AllLinux = 3,
+        LinuxArm = 4,
+        Linux386 = 8,
+        WindowsAmd64 = 16,
+        WindowsArm64 = 32
     }
 
-    public partial interface IDistributedApplicationPublisher
+    public sealed partial class DeploymentStateSection
     {
-        System.Threading.Tasks.Task PublishAsync(ApplicationModel.DistributedApplicationModel model, System.Threading.CancellationToken cancellationToken);
-    }
+        public DeploymentStateSection(string sectionName, System.Text.Json.Nodes.JsonObject? data, long version) { }
 
-    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPIPELINES001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
-    public partial interface IPublishingActivityReporter
-    {
-        System.Threading.Tasks.Task CompletePublishAsync(string? completionMessage = null, CompletionState? completionState = null, bool isDeploy = false, System.Threading.CancellationToken cancellationToken = default);
-        System.Threading.Tasks.Task<IPublishingStep> CreateStepAsync(string title, System.Threading.CancellationToken cancellationToken = default);
+        public System.Text.Json.Nodes.JsonObject Data { get { throw null; } }
+
+        public string SectionName { get { throw null; } }
+
+        public long Version { get { throw null; } set { } }
     }
 
-    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPIPELINES001", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
-    public partial interface IPublishingStep : System.IAsyncDisposable
+    [System.Diagnostics.CodeAnalysis.Experimental("ASPIREPIPELINES002", UrlFormat = "https://aka.ms/aspire/diagnostics/{0}")]
+    public partial interface IDeploymentStateManager
```

Comment:
It feels odd having these "Deployment" types under `namespace Aspire.Hosting.Publishing`